File: CC Contests/Pythagorean_Pair.py
from math import sqrt
from re import I


t = int(input())

while t:

    n = int(input())

    x = 0

    while(n%2 == 0):
        x += 1
        n /= 2
    
    if x%2 == 1:
        x -= 1
        n *= 2

    i = 0
    while i*i <= n:

        y = n-i*i
        z = int(sqrt(y))
        if z*z == y:
            print(z<<(x//2),end=" ")
            print(i<<(x//2))
            break

        i += 1
    

    t-=1

File: CC Contests/Pythagorean_Pair.cpp
#include <bits/stdc++.h>
using namespace std;
int n, x;
void solve()
{
    cin >> n;
    x = 0;
    while (n % 2 == 0)
    {
        x++;
        n /= 2;
    }
    if (x%2 == 1)
    {
        x--;
        n *= 2;
    }
    for (int i = 0; i*i <= n; ++i)
    {
        int y = n - i*i, z = sqrt(y);
        if (z*z == y)
        {
            cout << (z << (x / 2)) <<" "<<(i <<(x / 2)) << endl;
            return;
        }
    }
    puts("-1");
}
        int main()
        {

            int t;
            cin >> t;
            while (t--)
            {
                // CODE HERE
                solve();
            }
            return 0;
        }